### PR TITLE
remove str where not needed from alignment code

### PR DESCRIPTION
**Pull request by mdehoon**
- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit`` locally,
and understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

This PR removes calls to `str` where unnecessary from our alignment modules.

### CODE REVIEW

File: Bio/AlignIO/MauveIO.py [hunk 1]
```diff
@@ -160,12 +160,12 @@ def _write_record(self, record, record_idx=0):
         # of the produced XMFA file.
         if "start" in record.annotations and "end" in record.annotations:
             suffix0 = "/%s-%s" % (
-                str(record.annotations["start"]),
```
Comment (peterjc) [inline Bio/AlignIO/MauveIO.py:163]: Not sure about removing this - seems quite likely an end user might have recorded the start and end as integers?

**Review by peterjc — APPROVED** [review]
Query in ``Bio/AlignIO/MauveIO.py`` aside, this all looks fine.

The vast majority is using the ``Seq`` object equality directly (explicit str calls left over code from when we transitioned this from instance to string like equality).

File: Bio/AlignIO/MauveIO.py (hunk 1 above)
Comment (mdehoon) [reply Bio/AlignIO/MauveIO.py:163]: The `"%s"` will take care of that.

Comment (peterjc) [reply Bio/AlignIO/MauveIO.py:163]: So it will, thanks.

**Pull request merged**